src/features/island/collectibles/components/StoneFence.tsx

```diff
@@ -0,0 +1,118 @@
+import React from "react";
+
+import { GRID_WIDTH_PX } from "features/game/lib/constants";
+import { GameGrid } from "features/game/expansion/placeable/lib/makeGrid";
+
+import topAndRightEdge from "assets/decorations/stoneFence/top_right.png";
+import topAndLeftEdge from "assets/decorations/stoneFence/top_left.png";
+import bottomAndLeftEdge from "assets/decorations/stoneFence/bottom_left.png";
+import bottomAndRightEdge from "assets/decorations/stoneFence/bottom_right.png";
+import topAndLeftAndRightAndBottom from "assets/decorations/stoneFence/top_left_right_bottom.png";
+import topAndLeftAndRight from "assets/decorations/stoneFence/top_left_right.png";
+import bottomAndLeftAndRight from "assets/decorations/stoneFence/bottom_left_right.png";
+import rightTopAndBottom from "assets/decorations/stoneFence/right_top_bottom.png";
+import leftTopAndBottom from "assets/decorations/stoneFence/left_top_bottom.png";
+import bottom from "assets/decorations/stoneFence/bottom.png";
+
+import horizontalOne from "assets/decorations/stoneFence/x_1.png";
+import horizontalTwo from "assets/decorations/stoneFence/x_2.png";
+import horizontalThree from "assets/decorations/stoneFence/x_3.png";
+import horizontalFour from "assets/decorations/stoneFence/x_4.png";
+
+import verticalOne from "assets/decorations/stoneFence/y_1.png";
+import verticalTwo from "assets/decorations/stoneFence/y_2.png";
+import verticalThree from "assets/decorations/stoneFence/y_3.png";
+import verticalFour from "assets/decorations/stoneFence/y_2.png";
+
+type Edges = {
+  top: boolean;
+  right: boolean;
+  bottom: boolean;
+  left: boolean;
+};
+
+interface Props {
+  x: number;
+  y: number;
+  grid: GameGrid;
+}
+
+const horizontalImages = [
+  horizontalOne,
+  horizontalTwo,
+  horizontalThree,
+  horizontalFour,
+];
+
+const verticalImages = [verticalOne, verticalTwo, verticalThree, verticalFour];
+
+export const StoneFence: React.FC<Props> = ({ x, y, grid }) => {
+  const edges: Edges = {
```

Comment:
Recommend: Please update the PR description to include some images that show stone fence connecting with wood fence.

Question: This surface area of the logic below looks like it could be shrunk/optimized.  Do you want input on that or leave it as is?